### PR TITLE
clientv3: retry non-mutable rpcs on Internal codes

### PR DESCRIPTION
**Pull request by heyitsanthony**
Otherwise send errors will cancel the RPC via isHaltErr.

### CODE REVIEW

File: clientv3/client.go [hunk 1]
```diff
@@ -295,7 +295,13 @@ func isHaltErr(ctx context.Context, err error) bool {
...
 		return false
 	}
-	return grpc.Code(err) != codes.Unavailable
+	code := grpc.Code(err)
+	// Unavailable codes mean the system will be right back.
+	// (e.g., can't connect, lost leader)
+	// Internal codes seem to mean something failed, leaving the
+	// system in an inconsistent state, but retrying could make progress.
+	// (e.g., failed in middle of send, corrupted frame)
+	return code != codes.Unavailable && code != codes.Internal
```
Comment (xiang90) [inline clientv3/client.go]: probably add a todo here. i do not really like the work `seem to mean` here. Eventually, we should have confidence on error handling.

**Comment by xiang90** [comment]
lgtm

**Comment by gyuho** [comment]
This should fix https://github.com/coreos/etcd/issues/6277?

**Comment by heyitsanthony** [comment]
@gyuho yes, it's tagged in one of the patches

**Pull request merged**